test/rxd/3d/test_nodes_update.py

```diff
@@ -0,0 +1,39 @@
+import pytest
+
+@pytest.fixture
+def ics_example(neuron_instance):
+    """A simple 3D model to test nseg changes"""
+
+    h, rxd, data, save_path = neuron_instance
+    rxd.set_solve_type(dimension=3)
+    soma = h.Section(name='soma')
+    soma.L = 5
+    soma.diam = 5
+    soma.nseg = 5
+    r = rxd.Region([soma], nrn_region='i', dx=0.5)
+    c = rxd.Species(r, initial=1.0)
+    model = (soma, r, c)
+    yield (neuron_instance, model)
+
+def test_nodes_update(ics_example):
+    """Test nseg changes node for a 3D species"""
+
+    (h, rxd, data, save_path), (soma, r, c) = ics_example
+    assert(len(set([nd.segment for nd in c.nodes])) == 5)
+    soma.nseg = 3
+    assert(len(set([nd.segment for nd in c.nodes])) == 3)
```

Comment:
Unfortunately this test also passes in NEURON 7.8.2 so this doesn't test what you want it to.

The issue is that when `nseg=3` then `soma(0.1) == soma(0.3)` and `soma(0.9) == soma(0.7)` so you still get three distinct segments whether or not it remapped.

You could also just use a set comprehension here instead of constructing a list and then turning it into a set:

    {nd.segment for nd in c.nodes}

Reply:
7.8.2 would fail if the test was instead

    assert(len({str(nd.segment) for nd in c.nodes}) == 3)